enum SendRecordType
{
    ACTIVE = 1;
    RESPONSE_REQUEST = 2;
}

enum ReceiveRecordType
{
    BY_SELF_REQUEST = 1;
    BY_OTHER_ACTIVE = 2;
}

enum NotificationType
{
    REQUEST_GIFT = 1;
    RECEIVE_GIFT = 2;
}

message GoodsInfo
{
    required uint32 id = 1;
    required uint32 count = 2;
}

message SendRecord
{
    required uint32 uin = 1;
    required SendRecordType type = 2;
}

message ReceiveRecord
{
    required uint32 uin = 1;
    required ReceiveRecordType type = 2;
}

message RequestRecord
{
    required uint32 uin = 1;
}

message Notification
{
    required uint32 uin = 1;
    required NotificationType type = 2;
    repeated GoodsInfo goods_list = 3;
}

message ExchangePresentsInfo
{
    repeated SendRecord send_records = 1;
    repeated ReceiveRecord receive_records = 2;
    repeated RequestRecord request_records = 3;
    repeated Notification notifications = 4;
    required uint32 uin = 5;
    required uint32 last_login_time = 6;
    required uint32 unprocessed_message_count = 7;
}
